FreeBSD arm64 compile fails due to type mismatch

Encountered while building the Deltachat-rpc-server:

```
error[E0308]: mismatched types
  --> /wrkdirs/usr/ports/net/deltachat-rpc-server/work/core-1.159.3/cargo-crates/netdev-0.31.0/src/interface/unix.rs:192:28
   |
192 |             let nlen: i8 = (*sa).sa_data[3];
   |                       --   ^^^^^^^^^^^^^^^^ expected `i8`, found `u8`
   |                       |
   |                       expected due to this
   |
help: you can convert a `u8` to an `i8` and panic if the converted value doesn't fit
   |
192 |             let nlen: i8 = (*sa).sa_data[3].try_into().unwrap();
   |                                            ++++++++++++++++++++

error[E0308]: mismatched types
  --> /wrkdirs/usr/ports/net/deltachat-rpc-server/work/core-1.159.3/cargo-crates/netdev-0.31.0/src/interface/unix.rs:193:28
   |
193 |             let alen: i8 = (*sa).sa_data[4];
   |                       --   ^^^^^^^^^^^^^^^^ expected `i8`, found `u8`
   |                       |
   |                       expected due to this
   |
help: you can convert a `u8` to an `i8` and panic if the converted value doesn't fit
   |
193 |             let alen: i8 = (*sa).sa_data[4].try_into().unwrap();
   |                                            ++++++++++++++++++++
```


The arm64 API defines char as unsigned which I believe is the root cause here: https://github.com/ARM-software/abi-aa/blob/main/aapcs64/aapcs64.rst#10arm-c-and-c-language-mappings

--- a/src/interface/unix.rs
+++ b/src/interface/unix.rs
@@ -229,8 +229,8 @@ fn sockaddr_to_network_addr(
         if sa.is_null() {
             (None, None, None)
         } else if (*sa).sa_family as libc::c_int == libc::AF_LINK {
-            let nlen: i8 = (*sa).sa_data[3];
-            let alen: i8 = (*sa).sa_data[4];
+            let nlen: i8 = (*sa).sa_data[3] as i8;
+            let alen: i8 = (*sa).sa_data[4] as i8;
             if alen > 0 && alen as u8 + nlen as u8 + 8 <= (*sa).sa_len {
                 let ptr = (*sa).sa_data.as_mut_ptr();
                 let extended =

